Docker.DotNet/DockerPipeStream.cs

```diff
@@ -0,0 +1,124 @@
+using System;
+using System.IO;
+using System.IO.Pipes;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Net.Http.Client;
+using Microsoft.Win32.SafeHandles;
+
+namespace Docker.DotNet
+{
+    internal class DockerPipeStream : WriteClosableStream
```

Comment:
Go has taken over your life.. `WriteClosableStream`. Should we post something like this to CoreFX as feature request and see what they say? I agree with its merits but would be interested to see what the core team thinks about how to bring this type of patterns to C#